src/PostgrestQueryBuilder.ts

```diff
@@ -113,24 +113,31 @@ export default class PostgrestQueryBuilder<
    *
    * `"estimated"`: Uses exact count for low numbers and planned count for high
    * numbers.
+   *
+   * @param options.defaultToNull - Make missing fields default to `null`.
+   * Otherwise, use the default value for the column.
    */
   insert<Row extends Relation extends { Insert: unknown } ? Relation['Insert'] : never>(
     values: Row | Row[],
     {
       count,
+      defaultToNull = true,
```

Comment:
@dshukertjr As Vinzent said, there's no breaking change.

Reply:
Ah, I see that this is only applicable when you are inserting multiple rows in bulk huh? I was thinking it applies to when you are inserting a single row as well. Maybe we could add a note about that in the comments

Reply:
@soedirgo Right, so pretty much [when the inserted value is passed as an array](https://github.com/supabase/postgrest-js/blob/master/src/PostgrestQueryBuilder.ts#L148), the `defaultToNull` gets applied, but when the inserted value is an single object, the default value will be automatically be applied for missing fields. 

I think the current comments make it sound like even for single object insert, the columns that are left out will be null instead of having their default values, hence I thought it was a breaking change. I think there will be others that are also confused with the wording.

Reply:
> even for single object insert, the columns that are left out will be null instead of having their default values

Yes, this is the current behavior - regardless of single vs. bulk inserts, if you set `&columns=...` it'll use set `null`, and if you do set it it'll use default values. We always set `&columns=...`, so it's always the former

Reply:
>Yes, this is the current behavior - regardless of single vs. bulk inserts, if you set &columns=... it'll use set null, and if you do set it it'll use default values. We always set &columns=..., so it's always the former

@soedirgo 
Hmm, do you mean a situation like this, where I have a table with nullable columns with some default values and insert an object with some missing keys? When I perform the following, I do get the default value inserted with supabase-js v2.26.0.

```sql
create table if not exists test (
    id int generated by default as identity primary key,
    created_at timestamptz null default now(),
    content text null default 'foo'
);
```

```typescript
await supabase.from('test').insert({}); // a row is created with timestamp for now and content of `foo`
```

Reply:
Oh I think you're right, `columns` is only set when doing a bulk insert 😬 I'll add a comment about this and make the behavior consistent in v3. Thanks for the catch!

Reply:
So there is currently a difference in inserting a single row as an object and a single row in an array with length 1?

Reply:
It wasn't intentional, but yes

Reply:
I just tried that and I think they are equal. I ran the following code in the test db of `postgrest-js` and other than the actual value of `username` they are equal:

```ts
let res1 = await postgrest.from('users').insert({ username: 'bot1' }).select()
let res2 = await postgrest
  .from('users')
  .insert([{ username: 'bot2' }])
  .select()
console.log(res1)
console.log(res2)
```

Especially is `status` = `ONLINE`, which is the default value of that column. So in both cases the default column is used and not null.

I ran further test understood it now. The missing fields are only mapped to null if a field is missing in one row, but present in another, because then that column is listed in `&columns=`. To use the default value in that case as well, you have to use the `defaultToNull=false` flag. For missing fields in a bulk insert with only one row, all fields are listed in `&columns=` and therefore the missing fields are mapped to the default.
So there is a difference in fields missing in all rows and fields missing in only a proper subset.
I just want to really understand that to properly document and implement it in `postgrest-dart`.

Reply:
@Vinzent03 I think that's correct - so there isn't a difference between single row insert vs. bulk insert with 1 row after all, sorry for the back and forth.